### PR TITLE
add pipeline metrics

### PR DESCRIPTION
**Pull request by drunkirishcoder**

### CODE REVIEW

File: core/src/batch/send.rs [hunk 1]
```diff
@@ -1,21 +1,58 @@
...
+    pub fn new(name: String, batch: B, tx: Tx) -> Self {
+        Send { name, batch, tx }
+    }
+
+    #[cfg(feature = "metrics")]
     #[inline]
-    pub fn new(batch: B, tx: Tx) -> Self {
-        Send { batch, tx }
+    pub fn new(name: String, batch: B, tx: Tx) -> Self {
+        let done_counter = new_counter("processed", &name);
```
Comment (zeeshanlakhani) [inline core/src/batch/send.rs]: maybe done/processed -> `completed`?

File: core/src/runtime/mod.rs [hunk 2]
```diff
@@ -178,7 +178,7 @@ impl Runtime {
     /// `port` is the logical name that identifies the port. The `installer`
     /// is a closure that takes in a `PortQueue` and returns a `Pipeline`
     /// that will be spawned onto the thread executor.
-    pub fn add_pipeline_to_port<T: Future<Output = ()> + 'static, F>(
+    pub fn add_pipeline_to_port<T: Pipeline + 'static, F>(
```
Comment (zeeshanlakhani) [inline core/src/runtime/mod.rs]: better bound here for T, 🌮

File: core/src/batch/send.rs (hunk 1 above)
Comment (drunkirishcoder) [reply core/src/batch/send.rs]: renamed variable.

**Comment by zeeshanlakhani** [comment]
re-approved.

**Pull request merged**